tests/model_validation/tests.py

```diff
@@ -29,28 +22,77 @@ def test_models_validate(self):
         #       See: https://code.djangoproject.com/ticket/21375
         management.call_command("check", stdout=six.StringIO())
 
-    def test_model_signal(self):
-        unresolved_references = post_init.unresolved_references.copy()
-        post_init.connect(on_post_init, sender='missing-app.Model')
-        post_init.connect(OnPostInit(), sender='missing-app.Model')
+    @isolate_apps('django.contrib.auth', kwarg_name='apps')
+    def test_lazy_reference_checks(self, apps):
+
+        class DummyModel(models.Model):
+            author = models.ForeignKey('Author', models.CASCADE)
+
+            class Meta:
+                app_label = "model_validation"
 
-        errors = run_checks()
+        class DummyClass(object):
+            def __call__(self, **kwargs):
+                pass
+
+            def dummy_method(self):
+                pass
+
+        def dummy_function(*args, **kwargs):
+            pass
+
+        apps.lazy_model_operation(dummy_function, ('auth', 'imaginarymodel'))
+        apps.lazy_model_operation(dummy_function, ('fanciful_app', 'imaginarymodel'))
+
+        post_init.connect(dummy_function, sender='missing-app.Model', apps=apps)
```

Comment:
Actually these are never even connected, since the sender never exists, which also means we couldn't disconnect them properly anyway. I think it's OK as-is.

That raises a question – should `ModelSignal.disconnect()` use the lazy operation system as well? Currently (both before and after this patch) this happens:

```python
>>> import django
>>> from django.db import models
>>> from django.db.models.signals import post_init
>>> 
>>> django.setup()
>>> 
>>> def receiver(sender, **kwargs):
...     print("post_init sent by %r" % sender)
... 
>>> post_init.connect(receiver, sender='auth.SomeModel')
>>> post_init.disconnect(receiver, sender='auth.SomeModel')
False
>>> 
>>> class SomeModel(models.Model):
...     class Meta:
...         app_label = 'auth'
... 
>>> s = SomeModel()
post_init sent by <class 'SomeModel'>
>>> 
```

I'll open another PR about that.